### ISSUE TITLE
Expand this example when using Managed Identity!

### ISSUE THREAD

**Issue by alvipeo** [MicrosoftDocs/sql-docs#5799]
I've been trying to make it work with Managed Identity for hours and gave up. Please see [my question on SO](https://stackoverflow.com/q/65245840/2896495).

This is very important for people who really use Always Encrypted in Azure. Thanks!

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 01456910-0583-1787-b8a8-2aef66378eec
* Version Independent ID: b55ac9e0-bdd7-8e3d-19d2-540b47ca2e58
* Content: [Example demonstrating use of Azure Key Vault provider with Always Encrypted - SQL Server](https://docs.microsoft.com/en-us/sql/connect/ado-net/sql/azure-key-vault-example?view=sql-server-ver15)
* Content Source: [docs/connect/ado-net/sql/azure-key-vault-example.md](https://github.com/MicrosoftDocs/sql-docs/blob/live/docs/connect/ado-net/sql/azure-key-vault-example.md)
* Product: **sql**
* Technology: **connectivity**
* GitHub Login: @karinazhou
* Microsoft Alias: **v-jizho2**

**Comment by WilliamAntonRohm** [comment]
@alvipeo -- Alex, thank you for your question. 

@karinazhou -- Carina, please look into this issue.

cc @cheenamalhotra for awareness on related issue #5796

**Comment by karinazhou** [comment]
Hi @alvipeo ,

The [AzureKeyVault example](https://docs.microsoft.com/en-us/sql/connect/ado-net/sql/azure-key-vault-example?view=sql-server-ver15) is for the case when using the Microsoft.Data.SqlClient driver directly. The example is has the `Integrated Security=true;` in connection string which uses windows credential / kerberos credential. If you are going to use it with Managed Identity, you will need to specify the `Authentication=Active Directory Managed Identity` in the connection string. The information is from here:
https://docs.microsoft.com/en-us/sql/connect/ado-net/sql/azure-active-directory-authentication?view=sql-server-ver15#using-active-directory-managed-identity-authentication

From the code snippet on the StackOverflow, you refer to this page for the Managed Identity part:
https://docs.microsoft.com/en-us/azure/app-service/app-service-web-tutorial-connect-msi#modify-aspnet-core
This example uses Entity Framework which uses Microsoft.Data.SqlClient as its back end. There are some differences when you use MDS directly and when you use EF / EF Core. 

From the AzureKeyVault example, we are using Microsoft.IdentityModel.Clients.ActiveDirectory to get the AccessToken by providing clientID and clientSecret which has access to the AzureKeyVault. I see that you use a different library to get the AccessToken. You can compare the results from both to see whether they are the same. 

Thanks,
Karina

**Comment by alvipeo** [comment]
I do use Microsoft.Data.SqlClient driver directly. I don't have `Integrated Security=true;`.
I didn't know about 'Authentication=Active Directory Managed Identity', let me try that.

You see, the point is you (or whoever done it) made a documentation article about *everything*, it a mix of everything and nothing. It's hard to follow, hard to extract the info for any particular case. Please follow the KISS principle!

**Comment by cheenamalhotra** [comment]
Issue is followed in https://github.com/dotnet/SqlClient/issues/847, you may close this ticket here.
